### DISCUSSION TITLE
External Scaler - scale to 0

### DISCUSSION THREAD

**Discussion by rumstead** [kedacore/keda#2847] [Q&A / Need Help] [answered]
I am still doing my due diligence and reading but I found something in the docs I believe is incorrect.

I was looking at the [external scaler interface](https://keda.sh/docs/2.6/concepts/external-scalers/#built-in-scalers-interface) and the `isActive` description reads: 

> IsActive is called on pollingInterval defined in the ScaledObject/ScaledJob CRDs. KEDA will scale to 1 if this method returns true.

Wouldn't it actually scale to whatever is defined in the `minReplicaCount`, defined in the `ScaledObject`/`ScaledJob` CRDs?

https://keda.sh/docs/2.6/concepts/scaling-deployments/#scaledobject-spec

EDIT: https://github.com/kedacore/keda-docs/pull/739/files

**Comment by zroubalik** [comment, marked as answer]
Well actually if `isActive` returns `false` the workload is scaled ot `minReplicaCount` (if optional `idleReplicaCount` is defined, then  the workload is scaled to `idleReplicaCount`).

**Reply by rumstead** [reply]
I see.
```
idleReplicaCount: 0                                # Optional. Default: ignored, must be less than minReplicaCount
minReplicaCount:  1                                # Optional. Default: 0
```
https://keda.sh/docs/2.6/concepts/scaling-deployments/#scaledobject-spec


My understanding is if `isActive` is true, it should scale the deployment to whatever is returned by `GetMetricSpec` and is limited by `maxReplicaCount` on the `ScaledObject`.

If `isActive` is false, it will scale to `minReplicaCount` on the `ScaledObject` or optionally `idleReplicaCount`.
https://keda.sh/docs/2.6/concepts/scaling-deployments/#idlereplicacount

So it seems the docs are incorrect for `isActive` in the [external scaler interface](https://keda.sh/docs/2.6/concepts/external-scalers/#built-in-scalers-interface)?

**Reply by zroubalik** [reply]
In fact it does. `idle` is optional. If it is not set, it is ignored. you can set `minReplicaCount` to 0 and if `isActive()==false` the replicas are scaled to 0.

**Reply by zroubalik** [reply]
But definitely open to contributions, to make the docs more understandable :)

**Reply by rumstead** [reply]
Sorry I edited underneath you. Happy to contribute it, if you agree. 

It seems the docs are incorrect for isActive in the [external scaler interface](https://keda.sh/docs/2.6/concepts/external-scalers/#built-in-scalers-interface)?

**Reply by rumstead** [reply]
Let me know what you think of this: https://github.com/kedacore/keda-docs/pull/739

**Reply by zroubalik** [reply]
Looking good!

**Reply by avgdorom** [reply]
@zroubalik I'm trying to wrap my head around external scalers, and had a thought -  (maybe a dumb thought, but I have to ask :-) )
If I have a service, that's implementing your ExternalScaler service contract, and it gets scaled down to 0 at some point, what will happen? I mean, the service is now down. Keda calls IsActive(), but no one answers, right?
Am I missing something here? Is it best, in this case, to scale between 1 (min) and n (max)?

**Reply by rumstead** [reply]
The service that is being scaled shouldn’t be the one determining when to scale. Ie you would have a service that is the decision maker for other apps being scaled.

**Reply by zroubalik** [reply]
@rumstead is correct on this :)
